### DISCUSSION TITLE
Wondering your thoughts on the future of UmiJS? About the latest NextJS developments of React.

### DISCUSSION THREAD

**Discussion by xXAvoraXx** [umijs/umi#10847] [General]
Hi, I like using UmiJS, it has a clean hierarchy and a neat code flow. I'm wondering what your thoughts are about create-next-app, which has been keeping the frontend industry busy lately. UmiJS is not popular and well known in my area, I find it difficult to describe the advantages of UmiJS while the industry is generally leaning towards NextJS.

**Comment by xXAvoraXx** [comment]
++

**Comment by fz6m** [comment]
Most of the developers using Umi are Chinese, so recommended use Chinese to get more answers

**Comment by nitinreddy3** [comment]
你好，我喜欢使用 UmiJS，它有一个干净的层次结构和整洁的代码流程。我想知道您对 create-next-app 有何看法，它最近一直让前端行业变得忙碌。 UmiJS 在我所在的领域并不流行和知名，我发现很难描述 UmiJS 的优势，而业界普遍倾向于 NextJS。

**Reply by xXAvoraXx** [reply]
UmiJS is the best client side rendering solution as stated on its website, while NextJS is a good solution for server side rendering. If you have components that you need to create on the server side, NextJS will be a good solution for you. To give an example of a problem I have, I have a problem where I need to calculate the menus and components on the server side and present them to the user, menu components and client components may vary according to the role and nature of each user. I cannot solve this with UmiJS.

Yes, UmiJS is not known by anyone in my location, this is a PR problem, they teach NextJS for react beginners and this causes it to become more popular and even monopolized. But when we analyze both solutions, UmiJS solves many problems more easily than NextJS, for example i18n, routing and so on, and it is a great advantage to be able to develop plugins for umijs.
